### PR TITLE
[Impeller] bulk update fresh glyph atlas.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/152977

When creating a new glyph atlas texture, write glyphs in bulk. Additionally uses this operation to clear the glyph atlas, removing the need for the other blit pass.

### CODE REVIEW

File: impeller/typographer/backends/skia/typographer_context_skia.cc [hunk 1]
```diff
@@ -244,6 +245,68 @@ static void DrawGlyph(SkCanvas* canvas,
...
+    }
+
+    DrawGlyph(canvas, SkPoint::Make(pos.GetLeft(), pos.GetTop()),
+              pair.scaled_font, pair.glyph, bounds, pair.glyph.properties,
+              has_color);
+  }
+
+  // Writing to a malloc'd buffer and then copying to the staging buffers
+  // benchmarks as substantially faster on a number of Android devices.
+  BufferView buffer_view = host_buffer.Emplace(
```
Comment (chinmaygarde) [inline impeller/typographer/backends/skia/typographer_context_skia.cc:299]: You could reserve an arena in the host buffer itself and wrap that arena as an SkBitmap to reduce one copy. There is a HostBuffer::Emplace overload that takes a callback that will give you the pointer to the buffer.

**Review by chinmaygarde — APPROVED** [review]
LGTM but I think a copy can be avoided.

File: impeller/typographer/backends/skia/typographer_context_skia.cc (hunk 1 above)
Comment (jonahwilliams) [reply impeller/typographer/backends/skia/typographer_context_skia.cc:299]: If you recall from a while back: Skia is going to read from the buffer (even with BlendMode.src rendering) while rendering the glyphs, and the device buffers are not host read cached - so the performance tanks

Comment (chinmaygarde) [reply impeller/typographer/backends/skia/typographer_context_skia.cc:299]: Ah yeah, I remember now. Add a comment to that effect maybe?

**Pull request merged**